pkg/volume/configmap/configmap.go

```diff
@@ -192,26 +192,35 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return nil
 }
 
-func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
-	payload := make(map[string][]byte, len(configMap.Data))
+func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap, defaultMode *int32) (payload map[string][]byte, mode map[string]int32, err error) {
```

Comment:
what happened to the idea of a map[string]struct?  I really dislike parallel maps.